tests/drivers/spi/spi_loopback/Kconfig

```diff
@@ -6,3 +6,20 @@ source "Kconfig.zephyr"
 
 config SPI_LOOPBACK_MODE_LOOP
 	bool "Configure the SPI in LOOP mode, so that no extra wiring is needed"
+
+if SOC_SERIES_STM32H7X
+
+config SPI_LOOPBACK_16BITS_FRAMES
+	bool "Use 16 bits frames for tests"
+
+config DT_DEFINED_NOCACHE
+	bool "Enable this if nocache regions are defined in devicetree"
+
+if DT_DEFINED_NOCACHE
```

Comment:
Nocache buffers can be obtained with CONFIG_NOCACHE_MEMORY or by defining nocache regions in devicetree. Use these Kconfigs to detect the devicetree-defined nocache region and to obtain its name.

Reply:
We likely inherit this heterogeneity of configurations from the various evolutions about DMA and cache handling.
Maybe some of them don't make sense anymore and maintaining all these configurations will be costly in the end. At one point we'll have to revisit this, and minimize the number of possible configurations.
I guess you now have a good view on this, don't hesitate to propose a way forward whenever you feel appropriate (you don't have to do it yourself though).

Reply:
It depends on which path Zephyr wants to follow. If the `CONFIG_NOCACHE_MEMORY` strategy is to be discarded, then I would delete all code related to it in the driver, and do either manual cache coherency, or add an extra **mandatory** Kconfig to the STM32 SPI driver to specify which nocache area it is expected to use (and assume the user has declared such area as nocache in devicetree). The problem with this is that not all STM32 SPI drivers need to care about this, so possibly such section should be mandatory for SPI STM32 when the current arch. has DCACHE. There are Kconfigs to detect this already, so it won't be very difficult I think.

On the other hand,  if  `CONFIG_NOCACHE_MEMORY` is the way to go, I would modify the stm32 SPI Kconfig to always select this feature. Then, go to the driver's source code and remove any logic related to handle the possibility of it not being enabled, and stop taking into account other nocache areas declared in devicetree.

Finally, I would remove from the SPI loopback tests the code related to the option discarded above.

The main point is deciding which way to go. It seems to me the devicetree option is the way to go, because devicetree is the place to describe how the HW is, and the user should have freedom to decide where to put this or that nocache buffer. Although this option seems more complex than just enabling a Kconfig.

Reply:
I think the direction is to remove `CONFIG_NOCACHE_MEMORY`, but there may have some blockers right now, this is tbc. Anyway this is not directly related to your PR.
Let me put this point as resolved so it doesn't this PR to progress.